dataaudit.py

```diff
@@ -157,6 +190,131 @@ def count_categorical_outliers(series):
+        header_row = next(csvfile).strip('\n').split(delimiter)
+    data = pd.read_csv(filepath, encoding='utf-8', sep=delimiter)
+    return header_row, data
+
+
+
+def read_excel_file(filepath):
+    '''
+    Read Given excel file.
+    '''
+    wb = xlrd.open_workbook(filepath)
+    sheet = wb.sheet_by_index(0)
+    header_row = sheet.row_values(0)
+    data = pd.read_excel(filepath, encoding='utf-8')
+    return header_row, data
+
+
+def duplicate_datacolumns(data):
+    '''
+    Given a dataframe identify same columns based on dtype.
+    Input: dataframe
+         देश              city  product   sales  growth product.1  sales.1 growth.1
+        भारत         Hyderabad  Biscuit   866.1  -27.0%   Biscuit    866.1   -27.0%
+        भारत         Hyderabad  芯芯片片    26.4  -24.2%   芯芯片片     26.4   -24.2%
+        भारत         Hyderabad    Crème    38.3  -29.1%     Crème     38.3   -29.1%
+        भारत         Hyderabad     Eggs   513.7  -11.3%      Eggs    513.7   -11.3%
+        भारत         Bangalore  Biscuit    41.9  -40.2%   Biscuit     41.9   -40.2%
+        भारत         Bangalore  芯芯片片    52.2    6.4%   芯芯片片     52.2     6.4%
+    
+    In the above dataframe product, sales, growth column data is repeated. 
+    Return Duplicate column names as output
+
+    Output:
+        dups = ['sales', 'product', 'growth']
+    '''
+    groups = data.columns.to_series().groupby(data.dtypes).groups
+    dups = []
+    for t, v in groups.items():
+        cs = data[v].columns
+        vs = data[v]
+        lcs = len(cs)
+        for i in range(lcs):
+            for j in range(i+1, lcs):
+                fill_na_flag = False   
+                if t == np.float64 or t == np.int64:
+                    iv_nan= vs.iloc[:,i].index[vs.iloc[:,i].apply(np.isnan)].values.tolist()
+                    jv_nan= vs.iloc[:,j].index[vs.iloc[:,j].apply(np.isnan)].values.tolist()
+                    if np.array_equiv(iv_nan, jv_nan):
+                        fill_na_flag = True
+                if fill_na_flag:
+                    iv_series = vs.iloc[:,i].fillna(0)
+                    jv_series = vs.iloc[:,j].fillna(0)
+                else:
+                    iv_series = vs.iloc[:,i]
+                    jv_series = vs.iloc[:,j]
+                iv = iv_series.tolist()
+                jv = jv_series.tolist()
+                if np.array_equiv(iv, jv):
+                    dups.append(cs[i])
+                    break
+    return dups
+
+
+def check_order_id_continous(data):
+    '''
+    Given a dataframe identify continous order id.
+    '''
+    order_id_continous_columns = []
+    for column in data:
+        s_data = data[column]
+        if s_data.dtypes == np.float64 or s_data.dtypes == np.int64:
+            s_data.fillna(0, inplace=True)
+            s_data_diff = s_data.diff().reset_index(drop=True)
+            if len(s_data_diff.unique()) == 2:
+                order_id_continous_columns.append(column)
+    if len(order_id_continous_columns) > 0:
```

Comment:
How about

```
def is_plus_one(series):
    return pd.api.types.is_numeric_dtype(series) and series.diff().ne(1).sum() == 1

[col for col in df if is_plus_one(df[col])]
```

Reply:
@pratapvardhan difference of the sum may 1 or `n` but it should be consistent through out the column.
Example:

order_id
--
100
200
300
400

Reply:
these wortks @pratapvardhan thanks :)